src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,98 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { getPostComments, removePostComment } from '../../api/comments';
+import { getPostDetails } from '../../api/posts';
+import { Loader } from '../Loader';
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
 
-export const PostDetails: React.FC = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+type Props = {
+  selectedPostId: number,
+};
+
+export const PostDetails: React.FC<Props> = ({ selectedPostId }) => {
+  const [postDetails, setPostDetails] = useState<Post>();
+  const [postComments, setPostComments] = useState<PostComment[]>([]);
+  const [commentsHidden, setCommentsHidden] = useState(true);
+  const [postDetailsLoaded, setPostDetailsLoaded] = useState(false);
+
+  const loadComments = async () => {
+    const commentsFromServer = await getPostComments(selectedPostId);
+
+    setPostComments(commentsFromServer);
+  };
+
+  const loadPostDetails = async () => {
+    setPostDetailsLoaded(false);
+    const postsFromServer = await getPostDetails(selectedPostId);
+
+    setPostDetailsLoaded(true);
+    setPostDetails(postsFromServer);
+  };
+
+  useEffect(() => {
+    if (selectedPostId) {
+      loadPostDetails();
+      loadComments();
+    }
+  }, [selectedPostId]);
+
+  const removeComment = async (commentId: number) => {
+    await removePostComment(commentId);
+
+    loadComments();
+  };
+
+  return (
+    <div className="PostDetails">
+      <h2>Post details:</h2>
+
+      {postDetails && postDetailsLoaded
+        ? (
+          <section className="PostDetails__post">
+            <p>{postDetails.body}</p>
+          </section>
+        )
+        : <Loader />}
```

Comment:
![Screenshot 2022-02-09 at 16 20 36](https://user-images.githubusercontent.com/23210681/153229176-da681181-0090-4cea-9da0-3562be8b3bdd.png)
You shouldn't show interactive with comment before it loaded as for me